kimu/core/intersection_tool_lines.py

```diff
@@ -0,0 +1,72 @@
+from qgis.PyQt.QtCore import QVariant
+from qgis.core import (
+    QgsFeature,
+    QgsField,
+    QgsPointXY,
+    QgsProject,
+    QgsGeometry,
+    QgsVectorLayer,
+    QgsWkbTypes,
+)
+from qgis.utils import iface
+
+from ..qgis_plugin_tools.tools.custom_logging import setup_logger
+from ..qgis_plugin_tools.tools.i18n import tr
+from ..qgis_plugin_tools.tools.resources import plugin_name
+
+LOGGER = setup_logger(plugin_name())
+
+
+class IntersectionLines:
+
+    @staticmethod
+    def __check_valid_layer(layer: QgsVectorLayer) -> bool:
+        """Checks if layer is valid"""
+        if (
+            isinstance(layer, QgsVectorLayer)
+            and layer.isSpatial()
+            and layer.geometryType() == QgsWkbTypes.LineGeometry
+        ):
+            return True
+        return False
+
+    def run(self) -> None:
+        layer = iface.activeLayer()
+        if not self.__check_valid_layer(layer):
+            LOGGER.warning(tr("Please select a line layer"), extra={"details": ""})
+            return
+
+        if len(layer.selectedFeatures()) != 2:
+            LOGGER.warning(tr("Please select two line features"), extra={"details": ""})
+            return
+
+        vl = QgsVectorLayer("Point", "temp", "memory")
+        crs = layer.crs()
+        vl.setCrs(crs)
+        pr = vl.dataProvider()
+        pr.addAttributes([QgsField("xkoord", QVariant.Double), QgsField("ykoord", QVariant.Double)])
+        vl.updateFields()
+
+        l = list()
+
+        features = layer.selectedFeatures()
+        for feat in features:
```

Comment:
If I understand correctly, this implementation uses only the start and end points from the 2 selected lines? So in case of 2 zig-zagging linestrings that cross multiple times, only one intersection point is found, which may not be located on either of the lines?

If so, I suggest adding a check that the line does not contain multiple segments to avoid confusion about how this tool works.

Reply:
This is also where the `ExplodeLines` tool comes in handy, so best to keep it in this PR!

Reply:
Good idea! I should add a check that each of the selected line features consists of only one straight line, any combination of straight lines (as one line feature) are not accepted. The only problem is that ExplodeLines tool produces a point layer and this intersection tool takes line features as input..

Reply:
Ah, I see! Hm, maybe you could try modifying the explodelines tool so it explodes a single linestring with n>1 segments to a new layer with n different line features?